src/Illuminate/Mail/MailManager.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Illuminate\Mail;
+
+use Illuminate\Support\Arr;
+use InvalidArgumentException;
+
+class MailManager
+{
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * The active connection instances.
+     *
+     * @var array
+     */
+    protected $connections = [];
+
+    /**
+     * The custom connection resolvers.
+     *
+     * @var array
+     */
+    protected $extensions = [];
+
+    /**
+     * Create a new mail manager instance.
+     *
+     * @param  \Illuminate\Foundation\Application  $app
+     * @param  \Illuminate\Mail\TransportFactory   $factory
+     * @return void
+     */
+    public function __construct($app, TransportFactory $factory)
+    {
+        $this->app = $app;
+        $this->factory = $factory;
+    }
+
+    /**
+     * Get a mailer connection instance.
+     *
+     * @param  string|null  $name
+     * @return \Illuminate\Mail\Mailer
+     */
+    public function connection($name = null)
+    {
+        $name = $name ?: $this->getDefaultConnection();
+
+        // If we haven't created this connection, we'll create it
+        // based on the config provided in the application.
+        if (! isset($this->connections[$name])) {
+            $this->connections[$name] = $this->makeConnection($name);
+        }
+
+        return $this->connections[$name];
+    }
+
+    /**
+     * Make the mailer connection instance.
+     *
+     * @param  string  $name
+     * @return \Illuminate\Mail\Mailer
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function makeConnection($name)
```

Comment:
```PHP
protected function makeConnection(string $name)
```

Reply:
I've never been able to find Laravel's specific code style guidelines. Could you provide a link to where it prohibits scalar type hints?

Reply:
There's nothing official that explicitly prohibits them, they're just not used elsewhere in the framework (except in the case of extending a third party dependency that uses them). Generally, Laravel follows psr2 with a few exceptions, like spaces in docblocks, as referenced [here](https://laravel.com/docs/5.6/contributions#coding-style).

Reply:
But PSR-2 doesn't prohibit type-hinting primitives, so if the guideline is PSR-2, there's no specific reason to disallow it. If it is specifically disallowed, it should be mentioned in the link you just posted. If it's not specifically disallowed, there's no reason not to include them (type safety, in general, is a good thing).